src/sqlfluff/core/rules/analysis/select.py

```diff
@@ -30,6 +30,8 @@ def get_select_statement_info(
     # Iterate through all the references, both in the select clause, but also
     # potential others.
     sc = segment.get_child("select_clause")
+    if not sc:
```

Comment:
Do you know what type of query or statement caused this? Worth adding a comment to explain when this may happen?

Reply:
Added a comment. Though would say it's code coding practice not to assume something exists and handle it gracefully 😁

Reply:
Definitely! The comment is more for context, because sometimes there's a "whaaaa" 🤯 feeling when you see something like this.

You may have noticed with the functional API (`Segments` and its brethren), when something is missing, it just returns an empty collection that you can still chain additional function calls onto. That's very useful, because you can (if you choose) chain together a lot of function calls without requiring an explicit "do we have something" check at each step. Makes the code simpler to write and read.